core/che-core-api-core/src/main/java/org/eclipse/che/security/SHA512PasswordEncryptor.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.security;
+
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import com.google.common.primitives.Bytes;
+
+import java.nio.charset.Charset;
+import java.security.SecureRandom;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * SHA-512 based encryptor {@code hash = sha512(password + salt) + salt}.
+ *
+ * @author Yevhenii Voevodin
+ */
+public class SHA512PasswordEncryptor implements PasswordEncryptor {
```

Comment:
Well, first i created _classic_ `SHA512PasswordEncryptor`  but then found a lot of articles recommending bcrypt/scrypt/PBKDF2 as safer mechanisms for password encryption, so i implemented another `PBKDF2PasswordEncryptor` with sha-512 as pseudorandom function. For `PBKDF2PasswordEncryptor` ecnryption time depends on the count of the iterations but `SHA512PasswordEncryptor` works faster, so i decided to not remove the kt as it is useful in tests.
Notice time:
![image](https://cloud.githubusercontent.com/assets/4053805/16482649/61214cec-3eb9-11e6-873b-d6da6180e84c.png)